<?xml version="1.0" encoding="utf-8"?>

<launch>
    <arg name="node_start_delay" default="2.0" />

    <!-- #### YOLOv3 - 2D Object Detection #### -->
    <arg name="network_param_file" default="$(find darknet_ros)/config/yolov3.yaml" />
    <arg name="image" value="image_publisher" />

    <include file="$(find darknet_ros)/launch/yolo_v3.launch">
        <arg name="image" value="$(arg image)" />
    </include>

    <!-- #### Frustrum Pointnet - 3D Object Detection #### -->
    <param name="fpointnet_image_topic_sub"   value="/image_publisher" />
    <param name="fpointnet_depth_topic_sub"   value="/depth_publisher" />
    <param name="fpointnet_bbox_2d_topic_sub" value="/darknet_ros/bounding_boxes" />
    <param name="fpointnet_bbox_3d_topic_pub" value="/fpointnet_3d_prediction" />
    <param name="fpointnet_frustum_topic_pub" value="/fpointnet_frustum_pointclouds" />
    <node name="fpointnet_subscriber" pkg="frustum_pointnet" type="kitti_sub_node" output="screen" />

    <!-- #### TF static transform publisher #### -->
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 map base_link 100" />

    <!-- #### KITTI Image Publisher #### -->
    <node name="kitti_publisher" pkg="frustum_pointnet" type="kitti_pub_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />
    
</launch>